ocaml/xapi/xapi_vmss.ml

```diff
@@ -91,20 +91,20 @@ let set_schedule ~__context ~self ~value =
 	Db.VMSS.set_schedule ~__context ~self ~value
 
 let set_type ~__context ~self ~value =
-	(* Check VMs associated to VMSS supports the new snapshot type *)
 	assert_licensed ~__context;
-	if value = `snapshot_with_quiesce then
+	(* For VMSS snapshot_type=snapshot_with_quiesce, Check VMs supports the snapshot_with_quiesce *)
+	if value = `snapshot_with_quiesce then begin
 		Pool_features.assert_enabled ~__context ~f:Features.VSS;
-	let snapshot_type = Record_util.vmss_type_to_string value in
-	let vms = Db.VMSS.get_VMs ~__context ~self in
-	if vms <> [] then
-		List.iter (fun vm ->
-			let allowed_operations = Db.VM.get_allowed_operations ~__context ~self:vm in
-			if not (List.exists (fun ty -> (Record_util.vm_operation_to_string ty) = snapshot_type)
-				allowed_operations) then
-				raise (Api_errors.Server_error(Api_errors.operation_not_allowed,
-					[Ref.string_of vm ; "VM doesn't support snapshot_type " ^ snapshot_type]));
-		) vms;
+		let snapshot_type = Record_util.vmss_type_to_string value in
+		let vms = Db.VMSS.get_VMs ~__context ~self in
+		if vms <> [] then
```

Comment:
`List.iter` will do nothing on an empty list - there is no need to test for it. You can do:

```
Db.VMSS.get_VMs ~__context ~self
|> List.iter (fun vm ->
...
)